main.js

```diff
@@ -41,56 +44,107 @@ const drawBall = () => {
 };
 
 const updatePaddles = () => {
-  console.log(paddleOneSpeed);
-
-  paddleOneSpeed -= paddleOneSpeed * friction;
-
-  if (paddleOneSpeed > maxSpeed) {
-    paddleOneSpeed = maxSpeed;
-  }
-  if (paddleOneSpeed < -maxSpeed) {
-    paddleOneSpeed = -maxSpeed;
-  }
-  if (paddleTwoSpeed > maxSpeed) {
-    paddleTwoSpeed = maxSpeed;
-  }
-  if (paddleTwoSpeed < -maxSpeed) {
-    paddleTwoSpeed = -maxSpeed;
-  }
+  paddleOneSpeed -= friction * paddleOneSpeed;
+  paddleTwoSpeed -= friction * paddleTwoSpeed;
 
+  //if they are moving pretty slow, just stop them.
   if (Math.abs(paddleOneSpeed) < friction) {
     paddleOneSpeed = 0;
   }
-
   if (Math.abs(paddleTwoSpeed) < friction) {
     paddleTwoSpeed = 0;
   }
 
+  //add intertia from keypresses
+  if (paddleOneControls.up) {
+    paddleOneSpeed -= acceleration;
+  }
+  if (paddleOneControls.down) {
+    paddleOneSpeed += acceleration;
+  }
+  if (paddleTwoControls.up) {
+    paddleTwoSpeed -= acceleration;
+  }
+  if (paddleTwoControls.down) {
+    paddleTwoSpeed += acceleration;
+  }
+
   paddleOne.y += paddleOneSpeed;
   paddleTwo.y += paddleTwoSpeed;
 
+  //bounce off the top and bottom
   if (paddleOne.y < 0) {
     paddleOne.y = 0;
+    paddleOneSpeed *= -1;
   }
   if (paddleOne.y > canvas.height - paddleOne.height) {
     paddleOne.y = canvas.height - paddleOne.height;
+    paddleOneSpeed *= -1;
   }
   if (paddleTwo.y < 0) {
     paddleTwo.y = 0;
+    paddleTwoSpeed *= -1;
   }
   if (paddleTwo.y > canvas.height - paddleTwo.height) {
     paddleTwo.y = canvas.height - paddleTwo.height;
+    paddleTwoSpeed *= -1;
   }
 };
 
 const updateBall = () => {
   ball.x += ballSpeedHorizontal;
   ball.y += ballSpeedVertical;
-  if (ball.x < ball.radius || ball.x > canvas.width - ball.radius) {
-    ballSpeedHorizontal *= -1;
-  }
-  if (ball.y < ball.radius || ball.y > canvas.height - ball.radius) {
-    ballSpeedVertical *= -1;
+
+  //scoring
+  if (ball.x - border < ball.radius) {
+    ballSpeedHorizontal *= -1 - Math.random() / 50;
+    paddleOne.height -= 5;
+  }
+  if (ball.x + border > canvas.width - ball.radius) {
+    ballSpeedHorizontal *= -1 - Math.random() / 50;
+    paddleTwo.height -= 5;
+  }
+
+  //is the game over?
+  if (paddleOne.height === 0) {
```

Comment:
```suggestion
  if (paddleOne.height <= 0) {
```
I'd do a less than check here.

If you change the height of the paddle to be non multiple of 5, it would be easy to introduce a bug

Reply:
Done.